### PR TITLE
JNG-3902 Fix timestamp asstring

### PR DESCRIPTION
**Pull request by bencegelei**
<table>
<td>
  <a href="https://blackbelt.atlassian.net/browse/JNG-3902" title="JNG-3902" target="_blank">BugJNG-3902</a>  Timestamp.asString() produces wrong output
  </td></table>
  <br />
 

This PR contains the following changes:
- Tests re-enabled
- assertions added to timestamp tests
- `timestamp!asString()` test added

### CODE REVIEW

**Review by noherczeg — APPROVED** [review]

**Review by Madmat8 — APPROVED** [review]

**Review by gaborflorian — APPROVED** [review]

File: judo-runtime-core-jsl-itest/src/test/java/hu/blackbelt/judo/runtime/core/jsl/FunctionsTest.java [hunk 1]
```diff
@@ -496,52 +494,120 @@ public void testTime() {
...
-        OffsetDateTime minusAllReversed = timestampFunctions.getMinusAllReversed().orElseThrow();
-        OffsetDateTime minusDateReversed = timestampFunctions.getMinusDateReversed().orElseThrow();
+        assertEquals(OffsetDateTime.of(2021, 8, 28, 10, 10, 9, 999000000, ZoneOffset.UTC), timestampFunctions.getMinusAllReversed().orElseThrow());
+        assertEquals(OffsetDateTime.of(2021, 8, 28, 11, 11, 11, 0, ZoneOffset.UTC), timestampFunctions.getMinusDateReversed().orElseThrow());
+
+        assertEquals(OffsetDateTime.of(2022, 9, 29, 11, 11, 10, 999000000, ZoneOffset.UTC), timestampFunctions.getMinusMilliseconds().orElseThrow());
+    }
+
+    @Test
+    public void testTimestampAsString() {
```
Comment (fmagnucz) [inline judo-runtime-core-jsl-itest/src/test/java/hu/blackbelt/judo/runtime/core/jsl/FunctionsTest.java:624]: Create a new entity for this test case in the jsl model, please.

**Review by fmagnucz — CHANGES_REQUESTED** [review]

File: judo-runtime-core-jsl-itest/src/test/resources/model/Functions.jsl [hunk 2]
```diff
@@ -298,11 +298,61 @@ entity TimestampFunctions {
...
+    derived String timestamp14AsString => `2022-03-03T14:00:00.123-03:00`!asString();
+    derived String timestamp14AsString1 => self.timestamp14!asString();
+    derived String timestamp15AsString => `2022-03-03T14:00:00.123-03:03`!asString();
+    derived String timestamp15AsString1 => self.timestamp15!asString();
+    derived String timestamp16AsString => `2022-03-03T14:00:00.123Z`!asString();
+    derived String timestamp16AsString1 => self.timestamp16!asString();
+    derived String timestamp17AsString => `2022-03-03T14:00:00Z`!asString();
+    derived String timestamp17AsString1 => self.timestamp17!asString();
+
     derived String ownTimestamp1AsString => self.timestamp1!asString();
```
Comment (fmagnucz) [inline judo-runtime-core-jsl-itest/src/test/resources/model/Functions.jsl]: The ownTimestamp1AsString and ownTimestamp2AsString fields don't use anywhere in the FunctionsTest.java.

**Review by fmagnucz — CHANGES_REQUESTED** [review]

File: judo-runtime-core-jsl-itest/src/test/resources/model/Functions.jsl (hunk 2 above)
Comment (bencegelei) [reply judo-runtime-core-jsl-itest/src/test/resources/model/Functions.jsl]: oops. nice catch:D

File: judo-runtime-core-jsl-itest/src/test/java/hu/blackbelt/judo/runtime/core/jsl/FunctionsTest.java (hunk 1 above)
Comment (bencegelei) [reply judo-runtime-core-jsl-itest/src/test/java/hu/blackbelt/judo/runtime/core/jsl/FunctionsTest.java:624]: https://github.com/BlackBeltTechnology/judo-runtime-core-jsl/pull/146/commits/0d16858f45c94dfde6b5942079927e7bf2753e92

File: judo-runtime-core-jsl-itest/src/test/resources/model/Functions.jsl (hunk 2 above)
Comment (bencegelei) [reply judo-runtime-core-jsl-itest/src/test/resources/model/Functions.jsl]: `ownTimestamps` ~removed~ no longer needed

**Review by robertcsakany — APPROVED** [review]

**Review by noherczeg — APPROVED** [review]

**Pull request merged**